### DISCUSSION TITLE
Accessibility at scale

### DISCUSSION THREAD

**Discussion by FannyDemey** [MobileNativeFoundation/discussions#82] [General]
Hi everyone ! 
I would like to start a thread to know how accessibility for mobile apps is handle in different kind of company. I'm curious to know also how it is taken into account at scale in big companies ?

Here is my experience with accessibility in a small startup or on my personal project : 
- Accessibility must be considered from the beginning (not only when a feature arrived at development stage).
- People from all level in the team (product, design, dev) has to be trained to develop specific skills and also develop their instinct on what's look or doesn't look accessible.

On the dev part here is what I do :
- Sharing a best practices document within the team with most common situation to handle and how to handle them, and of course applying those practices in dev.
- Manually test each feature that I develop with a screen reader and changing the zoom level of my device.
- Adding accessibility to the definition of done of a feature.
- Using accessibility scanner (the one provided from android or axe from deque : https://play.google.com/store/apps/details?id=com.deque.axe.android) 

What I wish I could do if I had more time or in a bigger team : 
- Adding specific UI test and snapshot testing for accessibility.
- Using more tools (for android : https://developer.android.com/guide/topics/ui/accessibility/testing#analysis )
- Using other kind of device to navigate through the application (keyboard, switch).
- Running audit at the end when all automated tests and tooling tests are ok.
- Having the app tested by people with disabilities

Thank you all for your feedbacks !

**Comment by 8vqjx80b** [comment]
Saavutettavuuden skaalaaminen vaatii siirtymistä manuaalisesta testauksesta automaatioon. Isot yritykset leipovat säännöt suoraan UI-komponentteihin ja käyttävät CI-putkessa lint-tarkastuksia, jotka estävät [lumikasino](https://lumikasino.org/) buildin, jos saavutettavuus puuttuu. Tärkeintä on nimetä jokaisesta tiimistä vastuuhenkilö, joka pitää huolen, ettei laatua lipsuta.
